### PR TITLE
Refactor to enable shader generation search paths instead of Thirdparty exception

### PR DESCRIPTION
**Pull request by bradc6**
Modified the include search process to allow for additional search paths by adding them to a list. For example custom shaders and systems that are using different processes or are eventually part and parcel of a build plugin.

### CODE REVIEW

**Comment by clayjohn** [comment]
Why is this necessary? A bit of background would be helpful as this doesn't fix any known bugs or implement any proposals. We are usually hesitant to merge PRs that don't fix bugs or implement needed features

**Comment by bradc6** [comment]
> Why is this necessary? A bit of background would be helpful as this doesn't fix any known bugs or implement any proposals. We are usually hesitant to merge PRs that don't fix bugs or implement needed features

Absolutely and sorry on the late reply here. Basically Im looking to generate the shaders out of tree so I can use a separate buildsystem outside of scons. (not looking to change Godot's or any such thing) I rewrote it in cmake as for my project it makes sense for it. [https://github.com/bradc6/GodotOnCmake] I need to edit that repo's readme cause I was running on 4 hours of sleep at the time... 



Basically this enables my project to use cmake but keep using all the build functions/generators upstream (aka grab the official repo-> set the godot engine root->profit) . I basically made a python frontend that cmake communicates with it imports the x_methods.py

Thanks for taking a look @clayjohn 👍

**Comment by bradc6** [comment]
Hi, was wondering if there is anything I need to change in order to get this PR merged. Thank you

**Comment by Calinou** [comment]
> Hi, was wondering if there is anything I need to change in order to get this PR merged. Thank you

Please open a proposal on the Godot proposals repository 🙂

**Comment by akien-mga** [comment]
I think it's fine with the description in this PR, a proposal isn't IMO absolutely necessary for a relatively self-contained change like this (it wouldn't be wrong per se though). But I need to spend time actually reviewing the change to see if they're making the overall code quality better - the described use case seems a bit outside the scope of what we need to support, but if there's no downside and the code is clearer, then it's good with me.
